continuum/generators/scenarios_generator.py

```diff
@@ -0,0 +1,63 @@
+import abc
+from continuum.scenarios import _BaseScenario, create_subscenario
+import torch
+import numpy as np
+
+
+class _BaseGenerator(abc.ABC):
+    """Abstract loader.
+
+    DO NOT INSTANTIATE THIS CLASS.
+
+    :param scenario: A Continuum scenarios
+    """
+
+    def __init__(
+            self,
+            scenario: _BaseScenario
+    ) -> None:
+        self.base_scenario = scenario
```

Comment:
Do we accept an instantiated scenario or the base class? Because in some situation (like sorting per color) we may want to mix the data of different task ids

Reply:
An instantiated scenario, should I make an union all types of scenarios for that?

Reply:
I'm not talking directly about the Typing, but rather the user experience: it will always be an instanciated scenario? or it could be a scenario + a dataset?

Because I'm not sure how you could do some split (like color) in the former

Reply:
maybe scenario + dataset is more general, and makes possible more types of generator

Reply:
I agree

Reply:
For ` ClassOrderGenerator `  and ` `TaskOrderGenerator `  it is a instantiate scenario to not have to forward all the parameters. But other generators might be used differently.